arch/x86/kvm/vmx/pkvm/hyp/io_emulate.c

```diff
@@ -191,3 +208,167 @@ int register_host_mmio_handler(unsigned long start, unsigned long end,
 
 	return ret;
 }
+
+/*
+ * mmcfg access in x86 only use simple mov instrcutions. So keep the decoder
+ * simple for now.
+ * TODO: make the decoder complete
```

Comment:
Hi Dmytro, sorry for late reply. Here we do emulation because we don't want host to access MMCFG region directly. If we do PV, then host is still able to access the MMIO without pkvm audit. What do you think about it?

Reply:
Seemed we need continue the old way by simple instruction decode, @semihalf-maluka-dmytro, if you agree, pls check if there is other comments for this series patch, if no, we will merge this PR.

Reply:
Thanks for the explanation. Yeah, I think if we can use the PCI_Config opregion it would indeed be a cleaner approach. If we have to change the ACPICA code, it would be precarious.

Reply:
Yep, I agree. The PR looks good to me.